crate.ts

```diff
@@ -83,9 +87,27 @@ export class Crate {
 
   async setVersion(version: string) {
     $.logStep(`Setting ${this.name} to ${version}...`);
-    for (const crate of this.repo.crates) {
-      await crate.setDependencyVersion(this.name, version);
+
+    const metadata = await getCargoMetadata(this.repo.folderPath);
+    const rootpath = $.path.join(metadata.workspace_root, "Cargo.toml");
+
+    if (this.repo.folderPath === metadata.workspace_root) {
```

Comment:
Loading the metadata each time might be slow. I think it's a safe assumption that the repo root will always contain the workspace Cargo.toml in our repos, so let's just do `if (this.repo.folderPath !== this.folderPath)` (sorry, i said `===` in the chat, but i think this should be `!==`).